src/main/java/net/earthcomputer/multiconnect/protocols/generic/ChunkDataTranslator.java

```diff
@@ -124,6 +121,42 @@ public static void submit(ChunkDataS2CPacket packet) {
                 buf.readTopLevelType(ChunkData.class, userData);
                 ChunkData chunkData = ChunkData.read(dimension.getMinimumY(), dimension.getMinimumY() + dimension.getHeight() - 1, userData, buf);
 
+                if (!isFullChunk) {
+                    List<ChunkDeltaUpdateS2CPacket> deltaUpdatePackets = new ArrayList<>();
+                    for (ChunkSection section : chunkData.getSections()) {
+                        if (section == null) continue;
+
+                        ShortSet positions = new ShortOpenHashSet();
+                        BlockPos.Mutable mutable = new BlockPos.Mutable();
+                        for (int x = 0; x < 16; x++) {
+                            for (int y = 0; y < 16; y++) {
+                                for (int z = 0; z < 16; z++) {
+                                    positions.add(ChunkSectionPos.packLocal(mutable.set(x, y, z)));
+                                }
+                            }
+                        }
```

Comment:
this could definitely be optimized with a custom implementation of ShortSet, but good enough for now.